tests/test_api.py

```diff
@@ -458,6 +459,67 @@ def test_create_specification_auth(testclient):
     assert r.data.namespace.name == namespace
 
 
+# Only testing size values that will always cause errors. Smaller values could
+# cause errors as well, but would be flaky since the test conda-store state
+# directory might have different lengths on different systems, for instance,
+# due to different username lengths.
+@pytest.mark.parametrize(
+    "size",
+    [
+        # minio.error.MinioException: S3 operation failed; code:
+        # XMinioInvalidObjectName, message: Object name contains unsupported
+        # characters.
+        # The error message is misleading: it's a size issue.
+        255,
+        # SQL error: value too long for type character varying(255)
+        256,
+    ],
+)
+def test_create_specification_auth_env_name_too_long(testclient, size):
+    namespace = "default"
+    environment_name = 'A' * size
+
+    testclient.login()
+    response = testclient.post(
+        "api/v1/specification",
+        json={
+            "namespace": namespace,
+            "specification": json.dumps({"name": environment_name}),
+        },
+    )
+    if size > 255:
+        assert response.status_code == 500
+        return  # error, nothing to do
```

Comment:
Note the change here compared to `test_server.py`. Again, this is due to different backends being used for these.